tests/integration-tests/tests/scheduler_plugin/test_scheduler_plugin.py

```diff
@@ -533,3 +535,46 @@ def _test_cluster_update(cluster, cluster_config):
     cluster.update(str(cluster_config), force_update="true")
     cluster.start()
     _check_fleet_status(cluster, "RUNNING")
+
+
+def _test_invoke_scheduler_plugin_event_handler_script(command_executor, compute_node, run_as_user):
+    """Test invoke-scheduler-plugin-event-handler.sh."""
+    # Create a dummy cluster config for test, echo "dummy config {Event} executed" during handler execution.
+    cluster_config = "/opt/parallelcluster/shared/cluster-config.yaml"
+    dummy_cluster_config = "/opt/parallelcluster/shared/dummy-cluster-config.yaml"
+    command_executor.run_remote_command(
+        f"sudo sed 's/echo \"/echo \"dummy config /g' {cluster_config} | sudo tee {dummy_cluster_config} >/dev/null"
+    )
+
+    compute_node_private_ip = compute_node.get("privateIpAddress")
+    for event in ["HeadInit", "HeadConfigure", "HeadFinalize"]:
+        command_executor.run_remote_command(
+            f"sudo invoke-scheduler-plugin-event-handler.sh --cluster-configuration {dummy_cluster_config}"
+            f" --event-name {event}"
+        ).stdout
+        head_scheduler_plugin_log_output = command_executor.run_remote_command(
+            f"cat {SCHEDULER_PLUGIN_LOG_OUT_PATH}"
+        ).stdout
+        assert_that(head_scheduler_plugin_log_output).contains(f"[{event}] - INFO: dummy config {event} executed")
+    for event in ["HeadClusterUpdate", "HeadComputeFleetUpdate"]:
+        command_executor.run_remote_command(
+            f"sudo invoke-scheduler-plugin-event-handler.sh --cluster-configuration {dummy_cluster_config}"
+            f" --event-name {event} --previous-cluster-configuration {cluster_config} "
+            "--computefleet-status START_REQUESTED"
+        ).stdout
+        head_scheduler_plugin_log_output = command_executor.run_remote_command(
+            f"cat {SCHEDULER_PLUGIN_LOG_OUT_PATH}"
+        ).stdout
+        if event == "HeadClusterUpdate":
+            assert_that(head_scheduler_plugin_log_output).contains(f"[{event}] - INFO: dummy config {event} executed")
+        else:
+            assert_that(head_scheduler_plugin_log_output).contains(f"[{event}] - INFO: {event} not supposed to be here")
```

Comment:
not related to this PR, but I think there is error here because log entry "INFO: {event} not supposed to be here" it's not supposed to be in the log, as it says

Reply:
I have fixed the compute fleet status format in https://github.com/aws/aws-parallelcluster-cookbook/pull/1369.
This is going to generate an error when calling `invoke-scheduler-plugin-event-handler.sh --event-name HeadComputeFleetUpdate` with the current command defined for the HeadComputeFleetUpdate event.

Reason is that it's now going to correctly enter the case
```
              if [[ "${STATUS}" == "START_REQUESTED" ]] && [[ ! -f "start_failure" ]]; then
                echo "HeadComputeFleetUpdate failing the first start execution"
                touch start_failure && exit 1
              fi
```
which is letting the event to fail.

To test the invoke script for this HeadComputeFleetUpdate event, I see two alternative paths:
* call the run_remote_command for HeadComputeFleetUpdate with raise_on_error set to false and check the log contains "dummy config HeadComputeFleetUpdate failing the first start execution"
* change the HeadComputeFleetUpdate ExecuteCommand to a command which doesn't fail and then check the log

Reply:
Changed the test accordingly. I removed `start_failure` file before execute the command, so it enters the case: "HeadComputeFleetUpdate failing the first start execution"